test/parallel/test-async-wrap-check-providers.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const crypto = require('crypto');
+const dgram = require('dgram');
+const dns = require('dns');
+const fs = require('fs');
+const net = require('net');
+const tls = require('tls');
+const zlib = require('zlib');
+const ChildProcess = require('child_process').ChildProcess;
+const StreamWrap = require('_stream_wrap').StreamWrap;
+const async_wrap = process.binding('async_wrap');
+const pkeys = Object.keys(async_wrap.Providers);
+
+const keyList = pkeys.slice();
+// Drop NONE
+keyList.splice(0, 1);
+
+
+function init(id) {
+  const idx = keyList.indexOf(pkeys[id]);
+  if (idx === -1)
+    return;
+  keyList.splice(idx, 1);
+}
```

Comment:
actually, I don't understand this change. it does a numeric comparison against a string comparison. For it to work similarly to how it should work before the change would have to be:
```js
function init(id) {
  keyList = keyList.filter(e => e != pkeys[id]);
}
```
Though still curious to why this approach is preferred.